src/redux/data.ts

```diff
@@ -784,6 +802,61 @@ export const transactionsReceived = (
   }
 };
 
+export const maybeFetchF2CHashForPendingTransactions = async ({
+  pendingTransactions,
+}: {
+  pendingTransactions: RainbowTransaction[];
+}) => {
+  const updatedPendingTransactions = await Promise.all(
+    pendingTransactions
+      .filter(tx => Boolean(tx.fiatProvider))
+      .map(async tx => {
+        switch (tx.fiatProvider?.name) {
+          case FiatProviderName.Ratio: {
+            loggr.debug(
+              `maybeFetchF2CHashForPendingTransactions`,
+              { provider: tx.fiatProvider?.name },
+              loggr.DebugContext.f2c
+            );
+
+            const { userId, orderId } = tx.fiatProvider;
+            const { data, error } = await gretch<ActivityItem>(
+              `https://f2c.rainbow.me/v1/providers/ratio/users/${userId}/activity/${orderId}`
+            ).json();
+
+            if (!data || error) {
+              loggr.error(
+                new RainbowError(
+                  `maybeFetchF2CHashForPendingTransactions: failed to fetch transaction data`
+                ),
+                {
+                  error,
+                  provider: tx.fiatProvider.name,
+                }
+              );
+            } else if (data.crypto.transactionHash) {
+              tx.hash = data.crypto.transactionHash;
+
+              analyticsV2.track(analyticsV2.event.f2cTransactionReceived, {
+                provider: FiatProviderName.Ratio,
+                sessionId: tx.fiatProvider.analyticsSessionId,
+              });
+            } else {
+              loggr.info(
+                `maybeFetchF2CHashForPendingTransactions: fetcher returned no transaction data`
+              );
```

Comment:
This isn't an error, it'll happen all the time if a deposit hasn't hit the blockchain yet.